### ISSUE TITLE
several questions about training

### ISSUE THREAD

**Issue by liangzaiPPD** [DulyHao/AlphaForge#6]
Question 1: Why is the loss_entropy coefficient set to 0? I tested and found that it doesn't work with one-hot matrices as input, requiring a probability distribution instead of one-hot. What is the meaning of loss_entropy?

Question 2: Why is it necessary to normalize the y of the trainer to 0-100?

Question 3: The calculation of the correlation coefficients between factors uses batch_ret, which is the regression coefficient of the target and x. Is there a special intention in using regression coefficients instead of direct correlation coefficients?

**Comment by DulyHao** [comment]
Q1： that was a test which has not been applied to the final version, please ignore that.
Q2:  To normalize y ensures it keeps stability of training net P, because as the zoo size grows up, the score space will get sparse, usually, high score factors will be less. So the gradient will be very small. Normalizing makes it "smooth"
Q3: Just for calculating efficiency, no other reasons.

**Comment by liangzaiPPD** [comment]
> Q3: Just for calculating efficiency, no other reasons.

I know the matrix(d,feature_size) used for calculating coefficients between features is for efficiency. But each cell value here is the result of batch_ret, could it be result of batch_pearsonr instead?

Thanks for your reply
Another question: Is it necessary to use a predictor network? Couldn't we directly train the generator using metric scores as y.
During train_net_p_with_weight, Is it possible for x that has already been trained to appear again during training? Could this cause the data in the validation set to have been trained on by the training set?

**Comment by DulyHao** [comment]
Oh, sorry for misunderstanding, for the Q3, this is a little trick from the industry experience, batch_ret is more efficient than batch_pearsonr when filterring the correalation. 

As for the importance of the predictor network, as we talked about this in our paper, the process to acquire y is not defferentiable which means it could not be directly trained. That is a black-box function, our whole paper is trying our best te indirectly to optimize this black-box function.

As for the problem of information leakage, X will repeatedly appear in the training set of network P, but this will not affect the overall training process and will not lead to information leakage, because all x was generated by training set data, and the parameter of net P will be randomized at each epoch.

In addition, the entire NetP plus NetG training all formulas and IC calculations are based on the data obtained from the training set before a certain date, there can be no future function. Our ultimate goal is to make sure that we train a good NetG and get a factor ZOO using the stock data before a certain point in time. The factor zoo will then be evaluated on the validation set and test set after that point in time.
